pydicom/filereader.py

```diff
@@ -284,11 +289,19 @@ def _is_implicit_vr(fp, implicit_vr_is_assumed, is_little_endian, stop_when):
     stop_when : None, optional
         Optional call_back function which can terminate reading.
         Needed to check if the next tag still belongs to the read dataset.
+    is_sequence : bool
+        True if called for a sequence, False for a top-level dataset.
 
     Returns
     -------
     True if implicit VR is used, False otherwise.
     """
+    # sequences do not switch from implicit to explicit encoding,
+    # but they are allowed to use implicit encoding if the dataset
+    # is encoded as explicit VR
+    if is_sequence and implicit_vr_is_assumed:
+        return implicit_vr_is_assumed
+
```

Comment:
Simpler with just `return True` - can't get there unless `implicit_vr_is_assumed` is True.

Reply:
I actually went forth and back with this one - wasn't sure what is better... changed it now.